### ISSUE TITLE
Compute the KL divergence

### ISSUE THREAD

**Issue by qiaojiim** [huggingface/trl#2854]
In the GRPO algorithm, the loss term related to KL divergence is calculated as follows: https://github.com/huggingface/trl/blob/main/trl/trainer/grpo_trainer.py#L704

`per_token_kl = torch.exp(ref_per_token_logps - per_token_logps) - (ref_per_token_logps - per_token_logps) - 1`

How can we interpret this calculation formula? This does not seem to be the same with the definition of KL divergence.

Is it an approximation of KL divergence?

**Comment by skandermoalla** [comment]
It's the estimator used by GRPO (ref eq 2 https://arxiv.org/pdf/2501.12948). For more details, you can check the `k3` estimator in this blogpost (http://joschu.net/blog/kl-approx.html).

**Comment by qiaojiim** [comment]
> It's the estimator used by GRPO (ref eq 2 https://arxiv.org/pdf/2501.12948). For more details, you can check the `k3` estimator in this blogpost (http://joschu.net/blog/kl-approx.html).

great
